galley/pkg/server/server.go

```diff
@@ -252,12 +256,19 @@ func (s *Server) Run() {
 	}
 }
 
+func (s *Server) getListener() net.Listener {
+	s.listenerMutex.Lock()
+	defer s.listenerMutex.Unlock()
+	return s.listener
+}
+
 // Address returns the Address of the MCP service.
 func (s *Server) Address() net.Addr {
-	if s.listener == nil {
+	l := s.getListener()
```

Comment:
I'm not sure private helper function is really needed here since you access the listener and mutex directly below. Something like the following might be sufficient. It also ensures that the `Addr()` is invoked with the lock held.

```golang
func (s *server) Address() net.Addr {
    s.listenerMutex.Lock()
    defer s.listenerMutex.Unlock()

    if s.listener == nil {
        return nil
    }
    return s.listener.Addr()
}
```

Reply:
Sure but it's ultimately the same thing. The accessor just puts the locking logic in one place.